src/app/modules/group/components/pending-request/pending-request.component.html

```diff
@@ -8,115 +8,122 @@
     <div
       *ngIf="showSwitch"
       style="flex: 1; flex-direction: row-reverse;display: flex;"
-      [ngClass]="{'all-disabled': status === 'loading'}">
+      [ngClass]="{'all-disabled': state === 'fetching'}">
       <alg-selection (change)="onSubgroupSwitch($event)" [items]="subgroupSwitchItems" style="width: 25rem;"></alg-selection>
     </div>
   </ng-template>
 
-  <div class="pending-request-container" *ngIf="status !== 'loading'; else loading" [ngSwitch]="status">
-    <alg-grid
-      *ngSwitchCase="'loaded'"
-      [selectedColumns]="columns"
-      [columns]="columns"
-      [data]="requests"
-      [groupInfo]="panel"
-      [(selection)]="selection"
-      (sort)="onCustomSort($event)"
-    >
-      <ng-template #headerTemplate let-columns>
-        <tr>
-          <th style="width: 3.3333rem; background-color: #d3dbe7;"></th>
-          <ng-container *ngFor="let col of columns">
-            <th
-              [pSortableColumn]="col.field"
-              [ngClass]="{'small-column' : ['at', 'group.name'].includes(col.field)}"
-              style="background-color: #d3dbe7;"
-            >
-              <div class="header-container">
-                {{ col.header }}
-                <p-sortIcon [field]="col.field"></p-sortIcon>
-              </div>
-            </th>
-          </ng-container>
-        </tr>
-      </ng-template>
-      <ng-template
-        #bodyTemplate
-        let-rowData
-        let-columns="columns"
-        let-rowIndex="rowIndex"
+
+  <p *ngIf="state === 'error';else noError" class="error-message">
+    <i class="fa fa-exclamation-triangle"></i>
+    Error while loading the pending requests.
+  </p>
+
+  <ng-template #noError>
+    <div class="pending-request-container" #noError>
+      <alg-grid
+        *ngIf="requests.length > 0"
+        [selectedColumns]="columns"
+        [columns]="columns"
+        [data]="requests"
+        [groupInfo]="panel"
+        [(selection)]="selection"
+        (sort)="onCustomSort($event)"
+        [loading]="state === 'fetching' || ongoingActivity !== 'none'"
```

Comment:
The two cases were separated on purpose to just block the UI while the action was ongoing and, only if it required a refresh, refresh it completely. We could consider that this new behavior is acceptable (even if not as good).

Reply:
fixed